core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts

```diff
@@ -129,13 +140,24 @@ angular.module('oppia').component('explorationSaveAndPublishButtons', {
         $scope.saveIsInProcess = false;
         $scope.publishIsInProcess = false;
         $scope.loadingDotsAreShown = false;
-
+        $scope.explorationCanBePublished = false;
         UserExplorationPermissionsService.getPermissionsAsync()
           .then(function(permissions) {
-            $scope.showPublishButton = function() {
-              return permissions.canPublish && $scope.isPrivate();
-            };
+            $scope.explorationCanBePublished = permissions.canPublish;
```

Comment:
Do we need `$rootScope.$applyAsync();` after this?

Reply:
No, but this was used everywhere else. since it is an async funtion

Reply:
Removed

Reply:
Sorry for the misunderstanding, I was asking whether we need applyAsync call here to update the UI? (I see applyAsync call in a sync function callbacks)

Reply:
Fixed it. DONE. Yep it should be there.